### PR TITLE
UnitTests: avoid using glob to find compilation units

### PR DESCRIPTION
**Pull request by shuffle2**
Using `*` was always a hack, and now VS issues a warning upon solution/project load, so let's avoid hitting bad behavior by explicitly listing files.

This change exposed some compiler warnings (apparently because some files were not being compiled on windows before?!), so the PR also fixes those.

Also fixes a bug in `IOS::HLE::FS::HostFileSystem::Rename` (comparing to invalidated vector iterator) introduced in a40f297d1d46255ac8e4655148738be82de672d2

### CODE REVIEW

**Comment by JosJuice** [comment]
@dolphin-emu-bot rebuild

**Comment by JosJuice** [comment]
@dolphin-emu-bot rebuild

File: Source/UnitTests/Core/IOS/ES/TestBinaryData.h [hunk 1]
```diff
@@ -8,8 +8,170 @@
 #include <cstddef>
 #include "Common/CommonTypes.h"
 
-extern const std::array<u8, 520> soup01_tmd;
-extern const std::array<u8, 108> soup01_tmd_view;
+constexpr std::array<u8, 520> soup01_tmd = {{
```
Comment (leoetlino) [inline Source/UnitTests/Core/IOS/ES/TestBinaryData.h:11]: Just wondering, what warning does this fix?

Comment (shuffle2) [reply Source/UnitTests/Core/IOS/ES/TestBinaryData.h:11]: `extern const` is not equivalent to the actual storage type (which is `constexpr`) and apparently [extern constexpr is not possible](https://stackoverflow.com/questions/30208685/how-to-declare-constexpr-extern)

Comment (shuffle2) [reply Source/UnitTests/Core/IOS/ES/TestBinaryData.h:11]: Perhaps the existing code could be used, msvc just requires another flag: https://docs.microsoft.com/en-us/cpp/build/reference/zc-externconstexpr?view=vs-2017

note: 
> The /permissive- option does not enable /Zc:externConstexpr.

File: Source/UnitTests/UnitTests.vcxproj [hunk 2]
```diff
@@ -59,16 +59,51 @@
...
+    <ClCompile Include="Common\Crypto\EcTest.cpp" />
+    <ClCompile Include="Common\EventTest.cpp" />
+    <ClCompile Include="Common\FixedSizeQueueTest.cpp" />
+    <ClCompile Include="Common\FlagTest.cpp" />
+    <ClCompile Include="Common\FloatUtilsTest.cpp" />
+    <ClCompile Include="Common\MathUtilTest.cpp" />
+    <ClCompile Include="Common\NandPathsTest.cpp" />
+    <ClCompile Include="Common\SPSCQueueTest.cpp" />
+    <ClCompile Include="Common\StringUtilTest.cpp" />
+    <ClCompile Include="Common\SwapTest.cpp" />
```
Comment (Rumi-Larry) [inline Source/UnitTests/UnitTests.vcxproj:88]: Extra line?

File: Source/UnitTests/UnitTests.vcxproj [hunk 3]
```diff
@@ -59,16 +59,51 @@
...
+    <ClCompile Include="Core\CoreTimingTest.cpp" />
+    <ClCompile Include="Core\DSP\DSPAcceleratorTest.cpp" />
+    <ClCompile Include="Core\DSP\DSPAssemblyTest.cpp" />
+    <ClCompile Include="Core\DSP\DSPTestBinary.cpp" />
+    <ClCompile Include="Core\DSP\DSPTestText.cpp" />
+    <ClCompile Include="Core\DSP\HermesBinary.cpp" />
+    <ClCompile Include="Core\IOS\ES\FormatsTest.cpp" />
+    <ClCompile Include="Core\IOS\FS\FileSystemTest.cpp" />
+    <ClCompile Include="Core\MMIOTest.cpp" />
+    <ClCompile Include="Core\PageFaultTest.cpp" />
```
Comment (Rumi-Larry) [inline Source/UnitTests/UnitTests.vcxproj:98]: ditto here

File: Source/UnitTests/UnitTests.vcxproj (hunk 2 above)
Comment (shuffle2) [reply Source/UnitTests/UnitTests.vcxproj:88]: what?

Comment (shuffle2) [reply Source/UnitTests/UnitTests.vcxproj:88]: i think you're indicating i should insert a blank line to delineate the difference between root dirs here? I would prefer not to.

File: Source/UnitTests/Core/IOS/ES/TestBinaryData.h (hunk 1 above)
Comment (leoetlino) [reply Source/UnitTests/Core/IOS/ES/TestBinaryData.h:11]: I think this is fine, it's not like we're embedding huge amounts of data. I wish [std::embed](http://www.open-std.org/jtc1/sc22/wg21/docs/papers/2018/p1040r0.html) was a thing though...

**Pull request merged**